tests/sender/send_data.py

```diff
@@ -315,3 +322,228 @@ def test_sender_as_handler_static(self):
             self.fail("Problems with test: %s" % str(error))
+
+    def test_config_files_path_standard_case(self):
+        """
+        Test that verifies that a correct path for the
+        configuration file is detected.
+        """
+
+        engine_config = SenderConfigSSL(address=(self.server, self.port),
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=True)
+        result = Sender.check_config_files_path(engine_config)
+        self.assertEqual(result, True)
+
+    def test_config_files_path_incorrect_key(self):
+        """
+        Test that verifies that an incorrect path for the
+        configuration raises an exception.
+        """
+        wrong_key = SenderConfigSSL(address=(self.server, self.port),
+                                    key="Incorrect key",
+                                    cert=self.cert,
+                                    chain=self.chain,
+                                    check_hostname=False,
+                                    verify_mode=CERT_NONE,
+                                    verify_config=True)
+
+        wrong_key_message = "Error in the configuration, " + \
+                            wrong_key.key + \
+                            " path does not exist"
+
+        wrong_cert = SenderConfigSSL(address=(self.server, self.port),
+                                     key=self.key,
+                                     cert="Incorrect cert",
+                                     chain=self.chain,
+                                     check_hostname=False,
+                                     verify_mode=CERT_NONE,
+                                     verify_config=True)
+
+        wrong_cert_message = "Error in the configuration, " + \
+                             wrong_cert.cert + \
+                             " path does not exist"
+
+        wrong_chain = SenderConfigSSL(address=(self.server, self.port),
+                                      key=self.chain,
+                                      cert=self.cert,
+                                      chain="Incorrect chain",
+                                      check_hostname=False,
+                                      verify_mode=CERT_NONE,
+                                      verify_config=True)
+        wrong_chain_message = "Error in the configuration, " + \
+                              wrong_chain.chain + \
+                              " path does not exist"
+        test_params = [
+            (wrong_key, wrong_key_message),
+            (wrong_cert, wrong_cert_message),
+            (wrong_chain, wrong_chain_message),
+        ]
+
+        for config, expected_result in test_params:
```

Comment:
**issue:** Each test case should only test one single thing. Here you are testing three different cases. Please, separate each case into a different test.